kdl-client/src/Pages/Server/pages/Server/pages/Project/pages/Tools/Tools.tsx

```diff
@@ -16,36 +16,80 @@ import VSCodeImg from './img/vscode.png';
 import cx from 'classnames';
 import styles from './Tools.module.scss';
 import { useState } from 'react';
+import useExternalBrowserWindows, {
+  channelName,
+} from './useExternalBrowserWindows';
+import { useParams } from 'react-router-dom';
+import { IpcMainEvent } from 'electron';
+
+export enum ToolTypes {
+  GITEA,
+  MINIO,
+  JUPYTER,
+  VSCODE,
+  DRONE,
+  MLFLOW,
+}
 
 type ToolProps = {
   img: string;
   title: string;
   description: string;
   disabled?: boolean;
+  onClick?: () => void;
 };
-const Tool: FC<ToolProps> = ({ img, title, description, disabled = false }) => (
-  <div className={cx(styles.cardContent, { [styles.disabled]: disabled })}>
+
+const Tool: FC<ToolProps> = ({
+  img,
+  title,
+  description,
+  disabled = false,
+  onClick = () => {},
+}) => (
+  <div
+    className={cx(styles.cardContent, { [styles.disabled]: disabled })}
+    onClick={() => (disabled ? () => {} : onClick())}
+  >
     <div className={styles.imgContainer}>
       <img className={styles.toolImg} src={img} alt={`${title}_img`} />
     </div>
     <p className={styles.toolTitle}>{title}</p>
     <p className={styles.toolDescription}>{description}</p>
   </div>
 );
+const { ipcMain } = require('electron').remote;
 
 function Tools() {
+  const { projectId } = useParams();
   const [active, setActive] = useState(false);
+  const { openWindow } = useExternalBrowserWindows();
   function toggleActive() {
     setActive(!active);
   }
 
+  const onMessage = (event: IpcMainEvent, args: any) => {
```

Comment:
event is unused, use '_' instead.
```suggestion
  const onMessage = (_: IpcMainEvent, args: any) => {
```